File: labrabota6-gr12-MoonLory/Lab6_1/src/main/java/com/example/lab6/MainActivity.java
package com.example.Lab6_1;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.Menu;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Locale;

public class MainActivity extends AppCompatActivity {
    TextView tvInfo;
    EditText etInput;
    Button bControl;

    int maxNumber = 200;
    int minNumber = 1;

    int guess;
    boolean gameFinished;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Locale myLocale = new Locale("ru");
        Resources res = getResources();
        DisplayMetrics dm = res.getDisplayMetrics();
        Configuration conf = res.getConfiguration();
        conf.locale = myLocale;
        res.updateConfiguration(conf, dm);

        tvInfo = (TextView) findViewById(R.id.textView2);
        etInput = (EditText) findViewById(R.id.editTextNumber);
        bControl = (Button) findViewById(R.id.button);

        guess = (int) (Math.random() * 100);
        gameFinished = false;

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    public void onClick(View v) {
        if (!gameFinished) {
            int inp;

            try {
                inp = Integer.parseInt(etInput.getText().toString());
            }catch (Exception e){
                tvInfo.setText((getResources().getString(R.string.invalidFormat)));
                etInput.setText("");
                return;
            }

            if(inp < minNumber || inp > maxNumber){
                tvInfo.setText((getResources().getString(R.string.outOfBorder)
                        + "(" + minNumber + ", " + maxNumber + ")"));
                etInput.setText("");
                return;
            }
            if (inp > guess)
                tvInfo.setText(getResources().getString(R.string.ahead));
            if (inp < guess)
                tvInfo.setText(getResources().getString(R.string.behind));
            if (inp == guess) {
                tvInfo.setText(getResources().getString(R.string.hit));
                bControl.setText(getResources().getString(R.string.play_more));
                gameFinished = true;
            }
        } else {
            guess = (int) (Math.random() * (maxNumber-minNumber) + minNumber);
            bControl.setText(getResources().getString(R.string.input_value));
            tvInfo.setText(getResources().getString(R.string.try_to_guess));
            gameFinished = false;
        }
        etInput.setText("");
    }
}

File: labrabota6-gr12-MoonLory/Lab6_2/src/main/java/com/example/lab6_2/MainActivity.kt
package com.example.lab6_2_2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.res.Configuration
import android.view.View
import java.util.*
import android.text.method.ScrollingMovementMethod
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private enum class SavedInstanceMainActivityStateEnum(val value: String) {
        infoTextView("CURRENT_TEXT_INFO_MAIN_ACTIVITY"),
        guessNumber("CURRENT_GUESS_NUMBER_MAIN_ACTIVITY"),
        isNumberGuessed("CURRENT_NUMBER_IS_GUESSED_MAIN_ACTIVITY"),
        buttonText("CURRENT_BUTTON_TEXT_MAIN_ACTIVITY"),
        countGuessing("CURRENT_COUNT_GUESSING_MAIN_ACTIVITY"),
        historyTextViewInfo("CURRENT_HISTORY_TEXT_VIEW_INFO_MAIN_ACTIVITY")
    }

    private var guessNumber = 0
    private var isNumberGuessed = false
    private var countGuessing = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        guessNumber = randomInt(1, 200)
        if (savedInstanceState != null) {
            restoreActivity(savedInstanceState)
        }

        button.setOnClickListener {
            if (!isNumberGuessed) {
                val str = editTextNumber.text.toString()
                if(str != "") {
                    val userInput = str.toInt()
                    editTextNumber.setText("")
                    countGuessing++
                    when {
                        userInput < 1 -> infoTextView.text =
                                resources.getString(R.string.error_less_than_1)
                        userInput > 200 -> infoTextView.text =
                                resources.getString(R.string.error_more_than_200)
                        userInput == guessNumber -> {
                            infoTextView.text = resources.getString(R.string.win)
                            isNumberGuessed = true
                            button.text = resources.getString(R.string.button_win_text)
                        }
                        userInput > guessNumber -> infoTextView.text =
                                resources.getString(R.string.error_more)
                        userInput < guessNumber -> infoTextView.text =
                                resources.getString(R.string.error_less)
                    }
                    historyTextViewInfo.text =
                            "${historyTextViewInfo.text}$countGuessing: $userInput \n"
                }

            } else {
                guessNumber = randomInt(1, 200)
                infoTextView.text = resources.getString(R.string.info_text_view_init)
                button.text = resources.getString(R.string.button_play_text)
                historyTextViewInfo.text = ""
                isNumberGuessed = false
                countGuessing = 0
            }
        }
        historyTextViewInfo.movementMethod = ScrollingMovementMethod()
    }

    private fun randomInt(start: Int, end: Int): Int {
        return Random().nextInt(end) + start
    }

    private fun restoreActivity(savedInstanceState: Bundle){
        guessNumber = savedInstanceState.getInt(SavedInstanceMainActivityStateEnum.guessNumber.value)
        infoTextView.text = savedInstanceState.getString(SavedInstanceMainActivityStateEnum.infoTextView.value)
        isNumberGuessed = savedInstanceState.getBoolean(SavedInstanceMainActivityStateEnum.isNumberGuessed.value)
        button.text = savedInstanceState.getString(SavedInstanceMainActivityStateEnum.buttonText.value)
        countGuessing = savedInstanceState.getInt(SavedInstanceMainActivityStateEnum.countGuessing.value)
        historyTextViewInfo.text = savedInstanceState.getString(SavedInstanceMainActivityStateEnum.historyTextViewInfo.value)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putInt(SavedInstanceMainActivityStateEnum.guessNumber.value, guessNumber)
        outState.putInt(SavedInstanceMainActivityStateEnum.countGuessing.value, countGuessing)
        outState.putBoolean(SavedInstanceMainActivityStateEnum.isNumberGuessed.value, isNumberGuessed)
        outState.putString(SavedInstanceMainActivityStateEnum.buttonText.value, button.text.toString())
        outState.putString(SavedInstanceMainActivityStateEnum.historyTextViewInfo.value, historyTextViewInfo.text.toString())
        outState.putString(SavedInstanceMainActivityStateEnum.infoTextView.value, infoTextView.text.toString())

        super.onSaveInstanceState(outState)
    }
}